packages/openapi3/src/decorators.ts

```diff
@@ -32,3 +50,69 @@ export const $oneOf: OneOfDecorator = (
 export function getOneOf(program: Program, entity: Type): boolean {
   return program.stateMap(oneOfKey).get(entity);
 }
+
+const [getTagsMetadata, setTagsMetadata] = unsafe_useStateMap<Type, OpenAPI3Tag[]>(
```

Comment:
can it be `unsafe_useStateMap<Type, {[name: string]: OpenAPI3Tag}>`? so you know when you adds dup-name tag